datagenerator/src/main/java/com/linkedin/restli/datagenerator/csharp/CSharpArray.java

```diff
@@ -57,6 +57,8 @@ public String getName(NameModifier modifier) {
         return "List<" + getElementType().getName(NameModifier.NONE) + ">";
       case DATAMAP_PARSE:
```

Comment:
When doing init with data map for collection type fields, the `DATAMAP_PARSE` name modifier gives the fully qualified type of the field's data map for dealing with a data map that is strongly typed. The `DATAMAP_SHALLOW` modifier gives the shallow type of each layer when parsing a data map that is weakly typed (deserialized from JSON). Both of these are needed if we are keeping both of these cases, so I can delete the `DATAMAP_PARSE` if we decide that we don't want to handle that case anymore.